src/helpers/recompute_zones.ts

```diff
@@ -353,7 +353,7 @@ function constructZonesFromProfiles(
     pendingZones = nextPendingZones;
   }
   mergedZone.push(...pendingZones);
-  return mergedZone;
+  return mergedZone as T[];
```

Comment:
Not really happy about that but it's mighty annoying to type...

The rationale is that, as far as I know, calling recomputeZones on only Zones (and no unbounded zones) will always return Zones. So it'd be nice if it was typed as such.

The fact that it's typed as returning UnboundedZones make it very annoying to use for simple cases where we only handle zones. Do you have any good ideas @laa-odoo ?

Reply:
Indeed, this is never supposed to happen: recompute non-UnboundedZones will always give non-UnBoundedZones
But it will also be complicated to type the mechanics of recomputeZone.... The best would be to force the typing within recomputeZone itself so that it doesn't bother anyone. (
imo I am for)